presto-main/src/main/java/com/facebook/presto/memory/ClusterMemoryManager.java

```diff
@@ -100,7 +104,7 @@
     private final Map<MemoryPoolId, List<Consumer<MemoryPoolInfo>>> changeListeners = new HashMap<>();
 
     @GuardedBy("this")
-    private final Map<MemoryPoolId, ClusterMemoryPool> pools = new HashMap<>();
+    private final Map<MemoryPoolId, ClusterMemoryPool> pools;
```

Comment:
we could make `MemoryPoolId` an enum and use EnumMap here

Reply:
I think it makes sense to rename `MemoryPoolId` to `MemoryPool` and make it an enum as @findepi suggests. That way you will also get rid of `POOLS` above.